### PR TITLE
Add support to issue UDA (RGB21)

### PR DESCRIPTION
**Pull request by crisdut**
After merge this RGB-WG/rgb-wallet/pull/51, I will bump the rgb-wallet version and turn this PR ready for review.

### CODE REVIEW

File: src/uda.rs [hunk 1]
```diff
@@ -65,9 +65,11 @@ pub fn uda_schema() -> SubSchema {
                 GS_NOMINAL => Occurrences::Once,
                 GS_CONTRACT => Occurrences::Once,
                 GS_TIMESTAMP => Occurrences::Once,
+                GS_TOKENS => Occurrences::NoneOrOnce,
+                GS_ATTACH => Occurrences::NoneOrOnce,                   
             },
             assignments: tiny_bmap! {
-                OS_ASSET => Occurrences::Once,
+                OS_ASSET => Occurrences::OnceOrMore,
```
Comment (dr-orlovsky) [inline src/uda.rs:72]: No, we can't do that in that simple way :)
See my general review note.
```suggestion
                OS_ASSET => Occurrences::Once,
```

File: src/uda.rs [hunk 2]
```diff
@@ -77,10 +79,10 @@ pub fn uda_schema() -> SubSchema {
                 metadata: Ty::<SemId>::UNIT.id(None),
                 globals: none!(),
                 inputs: tiny_bmap! {
-                    OS_ASSET => Occurrences::Once
+                    OS_ASSET => Occurrences::OnceOrMore
```
Comment (dr-orlovsky) [inline src/uda.rs:82]:
```suggestion
                    OS_ASSET => Occurrences::Once
```

File: src/uda.rs [hunk 3]
```diff
@@ -77,10 +79,10 @@ pub fn uda_schema() -> SubSchema {
...
                 },
                 assignments: tiny_bmap! {
-                    OS_ASSET => Occurrences::Once
+                    OS_ASSET => Occurrences::OnceOrMore
```
Comment (dr-orlovsky) [inline src/uda.rs:85]:
```suggestion
                    OS_ASSET => Occurrences::Once
```

**Review by dr-orlovsky — CHANGES_REQUESTED** [review]
Right now UDA scheme is safe only if it is used to create a contract with _one_ NFT token, non-fractionable. If you change it to support multiple tokens, you have to write a validation script in AluAssembly which checks that during the transfer operations each unique token transferred once and only once. This is non-trivial to do in a correct and secure way and will take a lot of time to audit, thus right now I made this simple single-token schema to be used ASAP and left more advanced cases for later.

**Review by dr-orlovsky — APPROVED** [review]
ACK https://github.com/RGB-WG/rgb-schemata/commit/2ce8bec5e547bc57e17543311c555fc453a721a1

File: Cargo.toml [hunk 4]
```diff
@@ -14,10 +14,17 @@ readme = "README.md"
 
 [dependencies]
 amplify = "4.0.0"
-strict_encoding = "2.2.0"
+strict_encoding = "2.1.2"
```
Comment (dr-orlovsky) [inline Cargo.toml:17]: BTW why this decrease?

**Comment by dr-orlovsky** [comment]
Is this ready for merging? It is still in the draft status

**Comment by crisdut** [comment]
> Is this ready for merging? It is still in the draft status

We need a new release to rgb-wallet.

**Review by dr-orlovsky — APPROVED** [review]
ACK f633b0384921267a5b6740e216f1d447364ac231

**Comment by dr-orlovsky** [comment]
We can merge it with patch as it is now. I'd like to have release when other bugs will be fixed

**Pull request merged**